app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt

```diff
@@ -0,0 +1,53 @@
+package au.com.gridstone.trainingkotlin
+
+import android.app.Application
+import au.com.gridstone.trainingkotlin.BuildConfig.BASAE_URL
+import au.com.gridstone.trainingkotlin.apiservices.PokemonService
+import au.com.gridstone.trainingkotlin.viewmodels.HomeViewModel
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.OkHttpClient
+import org.koin.android.ext.koin.androidContext
+import org.koin.core.context.GlobalContext.startKoin
+import org.koin.core.module.Module
+import org.koin.dsl.module
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+val applicationModule: Module = module {
+  single<Retrofit> {
+    Retrofit.Builder()
+      // .baseUrl(Constants.BASE_URL)
+      .baseUrl(BASAE_URL)
+      .client(get())
+      // Moshi maps JSON to classes
+      .addConverterFactory(MoshiConverterFactory.create(
+        Moshi.Builder()
+          .add(KotlinJsonAdapterFactory())
+          .build()))
```

Comment:
declare moshi as its own leaf in our dependency graph
```kotlin
single { 
  Moshi.Builder()
    .add(KotlinJsonAdapterFactory())
    .build()))
}

single<Retrofit> {
  Retrofit.Builder()
      .baseUrl(BASAE_URL)
      .client(get())
      .addConverterFactory(MoshiConverterFactory.create(get())
}      
```

Reply:
Thanks i ahd updated this with separate definition